### PR TITLE
LPD-44173 Allow user to localize picklist and multi-select picklist fields

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-44173](https://liferay.atlassian.net/browse/LPD-44173)

[LPD-44173]: https://liferay.atlassian.net/browse/LPD-44173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -2413,6 +2445,19 @@ public void testGetObjectEntries() throws Exception {
...
+				"en_US",
+				HashMapBuilder.put(
+					"key", _listTypeEntryKey2
+				).build()
+			).put(
+				"pt_BR",
+				HashMapBuilder.put(
+					"key", _listTypeEntryKey1
+				).build()
+			).build()
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2490]: Will this test fail if we change?
```
"localizedPicklistObjectFieldName_i18n",
	HashMapBuilder.put(
		"en_US",
		HashMapBuilder.put(
			"key", _listTypeEntryKey1
		).build()
	).put(
		"pt_BR",
		HashMapBuilder.put(
			"key", _listTypeEntryKey2
		).build()
```

If the test doesn't fail, I think Brian would prefer `_listTypeEntryKey1` with `en_US`

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -5247,6 +5309,41 @@ private void _assertFilteredObjectEntriesSize(
 			objectEntries.toString(), size, objectEntries.size());
 	}
 
+	private void _assertListTypeEntries(
+		List<Object> expectedListTypeEntries,
+		List<ListEntry> actualListTypeEntries) {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:5314]: Please sort the parameters.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -5247,6 +5309,41 @@ private void _assertFilteredObjectEntriesSize(
...
+
+		if (ListUtil.isEmpty(expectedListTypeEntries)) {
+			return;
+		}
+
+		for (Object expectedListTypeEntry : expectedListTypeEntries) {
+			Assert.assertTrue(
+				ListUtil.exists(
+					actualListTypeEntries,
+					listEntry -> {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:5324]: Can you rename `listEntry` to `actualListTypeEntry` ?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 4]
```diff
@@ -5247,6 +5309,41 @@ private void _assertFilteredObjectEntriesSize(
...
+								expectedListTypeEntry, listEntry.getKey());
+						}
+
+						return Objects.equals(expectedListTypeEntry, listEntry);
+					}));
+		}
+	}
+
+	private void _assertListTypeEntry(
+		Object expectedListTypeEntry, ListEntry actualListTypeEntry) {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:5336]: Please sort the parameters.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 5]
```diff
@@ -5247,6 +5309,41 @@ private void _assertFilteredObjectEntriesSize(
...
+			Assert.assertTrue(
+				ListUtil.exists(
+					actualListTypeEntries,
+					listEntry -> {
+						if (expectedListTypeEntry instanceof String) {
+							return Objects.equals(
+								expectedListTypeEntry, listEntry.getKey());
+						}
+
+						return Objects.equals(expectedListTypeEntry, listEntry);
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:5330]: We could try to reuse the `_assertListTypeEntry()` here :thinking: 
```
actualListTypeEntry -> {
    try {
        _assertListTypeEntry(actualListTypeEntry, expectedListTypeEntry);
        return true; 
    } catch (AssertionError assertionError) {
        return false; 
    }
}
```

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 6]
```diff
@@ -4872,22 +4921,41 @@ else if (Objects.equals(
 							String.valueOf(actualValues.get(entry.getKey())))));
 			}
 		}
+		else if (Objects.equals(
+					expectedEntry.getKey(),
+					"localizedMultiPicklistObjectFieldName")) {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:4926]: Curiosity: I did not find the condition that englobes the multiPicklistObjectFieldName. Is there a test?

Comment (marcelabc) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:4926]: I know that is not the goal of this PR.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi, so sorry about my delay in reviewing this PR :disappointed: 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 5 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: Done!

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 4 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: Done!

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 3 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: Done!

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 2 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: Done!

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 1 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2490]: Hey @marcelabc ,

Initially, I set it as you suggested (en_US -> _listTypeEntryKey1 / pt_BR -> _listTypeEntryKey2). However, in other scenarios, I needed to test it with different values, so I adjusted the order accordingly.

Let me know if you'd like me to change it.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 6 above)
Comment (pedro-oliveira446) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: Yes, there’s a test [here](https://github.com/pedro-oliveira446/liferay-portal/blob/73c469fc01680098a10f727ae4ece9da2203e976/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L3891-L3895). In this case, the value is passed as a ListEntry, which is the same type returned by the DTO, so there’s no need for a specific conditional to compare them.

Initially, I tried to follow this pattern, setting the localizedMultiPicklistObjectFieldName values as a ListEntry array. However, I encountered other challenges, and the test became significantly more complex. To simplify, I decided to keep the values as Arrays.asList(_listTypeEntryKey3, _listTypeEntryKey4).

Let me know if further clarification is needed.

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:report:100587788

**Comment by pedro-oliveira446** [comment]
ci:report:100576628

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Pull request closed without merging**